app/server/appsmith-git/src/main/resources/git.sh

```
@@ -25,19 +42,24 @@ git_clone() {
 # Uploads git repo to Redis as compressed archive
 git_upload() {
     local redis_key="$1"
     local redis_url="$2"
     local target_folder="$3"
+    local key_value_pair_key="$4"
 
     trap 'rm -rf "'"$target_folder"'"' EXIT ERR
 
     rm -f "$target_folder/.git/index.lock"
 
+    local branches=$(branches_to_json "$target_folder")
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)